apps/gallery/js/MetadataParser.js

```diff
@@ -500,60 +586,67 @@ var metadataParser = (function() {
         canvas.toBlob(function(blob) {
```

Comment:
Before 2.5, that drawImage() call would fail with an exception if there was not enough memory, and the catch clause would handle it. But apparently the spec does not allow drawImage() to throw anything. So gecko changed, and now drawImage fails silently, creating those black thumbnails.  Eventually we'll probably want to add code to try to detect that: draw some random pixels into the canvas, call drawImage, then check that the random pixels have been modified. If not, then the drawImage failed.